test/middlewares/verifyToken.test.js

```diff
@@ -0,0 +1,84 @@
+import models from '../../server/database/models';
+import { generateToken, checkId } from '../../server/helpers';
+
+const { verifyToken } = middlewares;
+const { User, Sessions } = models;
+
+chai.use(sinonChai);
+
+
+describe(
+  'verify token middleware', () => {
+    afterEach(() => {
+      if (User.findOne.restore) User.findOne.restore();
+      if (Sessions.findOne.restore) Sessions.findOne.restore();
+    });
+    it('return error if no token', async () => {
+      const request = { headers: {} };
+      const response = { status() {}, json() {} };
+      const next = sinon.spy();
+      const status = sinon.stub(response, 'status').returnsThis();
+      sinon.stub(response, 'json').returns({});
+
+      await verifyToken(request, response, next);
+      expect(status).to.calledWith(401);
+    });
+
+    it('return error for non existing user', async () => {
+      const token = generateToken(2);
+      const user = checkId(1);
+      const request = { headers: { authorization: token } };
+
+      sinon.stub(User, 'findOne').returns(false);
+      const response = { status() {}, json() {} };
+      const next = sinon.spy();
+      const status = sinon.stub(response, 'status').returnsThis();
+      sinon.stub(response, 'json').returns({});
+      await verifyToken(request, response, next);
+      expect(status).to.calledWith(404);
+    });
+
+    it('return error for expired token', async () => {
+      const token = generateToken(1);
+      const request = { headers: { authorization: token } };
+      sinon.stub(User, 'findOne').returns(true);
+      sinon.stub(Sessions, 'findOne').returns(false);
+      const response = { status() {}, json() {} };
+      const next = sinon.spy();
+      const status = sinon.stub(response, 'status').returnsThis();
+      sinon.stub(response, 'json').returns({});
+      await verifyToken(request, response, next);
+      expect(status).to.calledWith(440);
+    });
+
+    it('it should go to next', async () => {
+      const token = generateToken(1);
+      const request = { headers: { authorization: token } };
+      sinon.stub(User, 'findOne').returns(true);
+      sinon.stub(Sessions, 'findOne').returns(true);
+      const response = { status() {}, json() {} };
+      const next = sinon.spy();
+      sinon.stub(response, 'status').returnsThis();
+      sinon.stub(response, 'json').returns({});
+      await verifyToken(request, response, next);
+    });
+
+    it('it should go to next', async () => {
+      const request = { headers: { authorization: 'auth' } };
+      sinon.stub(User, 'findOne').returns(true);
+      sinon.stub(Sessions, 'findOne').rejects();
+      const response = { status() {}, json() {} };
+      const next = sinon.spy();
+      const status = sinon.stub(response, 'status').returnsThis();
+      sinon.stub(response, 'json').returns({});
+      await verifyToken(request, response, next);
+      expect(status).to.calledWith(401);
+    });
+  }
+);
+
+export default chai;
```

Comment:
Why are you exporting chai

Reply:
In order for the test file to be accessible on the index file, it is required